File: Simple_Linear_Regression/simple_linear_regression.py
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = (1.0/3), random_state = 0)

# Fitting Simple Linear Regression model to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

#Predicting Test set
y_pred = regressor.predict(X_test)

#Visualizing the training set results
plt.scatter(X_train,y_train, color = 'red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.title('Salary Prediction Plot (training set)')

#Visualizing the test data and results
plt.scatter(X_test,y_test, color = 'red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.scatter(X_test,y_pred, color = 'blue')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.title('Salary Prediction Plot (test set)')

File: Simple_Linear_Regression/simple_linear_regression.R
#Data Preprocessing

#Importing Dataset
dataset <- read.csv('Salary_Data.csv')
#dataset <- dataset[,2:3]

#Diving the dataset into training set and test set
library(caTools)
set.seed(123)
split <- sample.split(dataset$Salary, SplitRatio = (2/3))
training_set <- subset(dataset, split == TRUE)
test_set <- subset(dataset, split == FALSE)

#Feature Scaling
# training_set[,2:3] <- scale(training_set[,2:3])
# test_set[,2:3] <- scale(test_set[,2:3])


#Fitting Simple Linear Regression
regressor <- lm(formula = Salary ~ YearsExperience,
                data = training_set)
#summary(regressor)

#Prediction for the test set
y_pred <- predict(regressor,newdata = test_set)

#Visualize training set results
#install.packages(ggplot2)
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             color = 'red') +
  geom_line(aes(x=training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary Prediction - Training Set') +
  xlab('years of Experience') +
  ylab('Salary')

#Visualize test set results
#install.packages(ggplot2)
#library(ggplot2)
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             color = 'red') +
  geom_point(aes(x = test_set$YearsExperience, y = y_pred),
             color = 'blue') +
  geom_line(aes(x=training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary Prediction - Test Set') +
  xlab('years of Experience') +
  ylab('Salary')